File: panel/src/main/java/com/effective/android/panel/utils/cutShort/OfficialCutShort.java
package com.effective.android.panel.utils.cutShort;

import android.content.Context;
import android.os.Build;
import android.view.View;

/**
 * 官方刘海计算
 * https://developer.android.com/guide/topics/display-cutout
 * Created by yummyLau on 20-03-27
 * Email: yummyl.lau@gmail.com
 * blog: yummylau.com
 */
public class OfficialCutShort implements DeviceCutShort {

    public static final String VENDOR = "Officail";

    @Override
    public boolean hasCutShort(Context context) {
        return true;
    }

    @Override
    public boolean isCusShortVisible(Context context) {
        return true;
    }

    @Override
    public int getCurrentCutShortHeight(View view) {
        Context context = view.getContext();
        if(!isCusShortVisible(context)){
            return 0;
        }
        int displayCutoutHeight = 0;
        try {
            if (Build.VERSION.SDK_INT >= 28 && view.getRootWindowInsets() != null && view.getRootWindowInsets().getDisplayCutout() != null) {
                displayCutoutHeight = view.getRootWindowInsets().getDisplayCutout().getSafeInsetTop();
            }
            return displayCutoutHeight;
        } catch (Exception var3) {
            return 0;
        }
    }
}

File: panel-kotlin/src/main/java/com/effective/android/panel/utils/cutShort/OfficialCutShort.kt
package com.effective.android.panel.utils.cutShort

import android.content.Context
import android.os.Build
import android.view.View

/**
 * 官方刘海计算
 * https://developer.android.com/guide/topics/display-cutout
 * Created by yummyLau on 20-03-27
 * Email: yummyl.lau@gmail.com
 * blog: yummylau.com
 */
class OfficialCutShort : DeviceCutShort {

    override fun hasCutShort(context: Context): Boolean {
        return true
    }

    override fun isCusShortVisible(context: Context): Boolean {
        return true
    }

    override fun getCurrentCutShortHeight(view: View): Int {
        val context = view.context
        if (!isCusShortVisible(context)) {
            return 0
        }
        var displayCutoutHeight = 0
        return try {
            if (Build.VERSION.SDK_INT >= 28 && view.rootWindowInsets != null && view.rootWindowInsets.displayCutout != null) {
                displayCutoutHeight = view.rootWindowInsets.displayCutout!!.safeInsetTop
            }
            displayCutoutHeight
        } catch (var3: Exception) {
            0
        }
    }

    companion object {
        const val VENDOR = "Officail"
    }
}